internal/v4/api.go

```diff
@@ -1008,21 +1009,38 @@ func (h *ReqHandler) serveMacaroon(_ http.Header, _ *http.Request) (interface{},
 // GET /delegatable-macaroon
 // See https://github.com/juju/charmstore/blob/v4/docs/API.md#get-delegatable-macaroon
 func (h *ReqHandler) serveDelegatableMacaroon(_ http.Header, req *http.Request) (interface{}, error) {
+	decoder := json.NewDecoder(req.Body)
```

Comment:
This method should check that req.Method == "GET".
Also, GET requests should not have bodies. Please use a form parameter (id=...) instead.